--- git-svn    2008-06-04 23:58:59.000000000 -0400
+++ git-svn    2008-06-05 00:03:43.000000000 -0400
@@ -118,6 +118,10 @@
                       "Deprecated alias for ".
               "'$0 init -T<trunk> -b<branches> -t<tags>'",
               \%init_opts ],
+    diff => [\&cmd_diff,
+            'View the diff between the last git commit and last SVN commit',
+             {}],
+            
     dcommit => [ \&cmd_dcommit,
                  'Commit several diffs to merge with upstream',
             { 'merge|m|M' => \$_merge,
@@ -401,6 +405,29 @@
     print "Done committing ",scalar @revs," revisions to SVN\n";
     unlink $gs->{index};
 }
+sub cmd_diff {
+    my $head = shift;
+    $head ||= 'HEAD';
+    my @refs;
+    my ($url, $rev, $uuid, $gs) = working_head_info($head, \@refs);
+        unless ($gs) {
+                die "Unable to determine upstream SVN information from ",
+                    "$head history\n";
+        }
+        my $last_rev;
+        my ($linear_refs, $parents) = linearize_history($gs, \@refs);
+    while(1) {
+        my $d = shift @$linear_refs or last;
+                unless (defined $last_rev) {
+                        (undef, $last_rev, undef) = cmt_metadata("$d~1");
+                        unless (defined $last_rev) {
+                                fatal "Unable to extract revision information ",
+                                      "from commit $d~1";
+                        }
+                }
+        command_noisy('diff-tree', "$d~1", "$d", '-p');
+    }
+}
 
 sub cmd_dcommit {
     my $head = shift;
